doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md

```diff
@@ -0,0 +1,1069 @@
+
+## 3.4 SyncD
+No changes to SyncD
+
+## 3.5 SAI
+No changes to SAI APIs.
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+List of  Open-config YANG models required for FRR-BGP Unified Configuration and Management are,
+
+    1) openconfig-network-instance.yang
+
+    2) openconfig-routing-policy.yang
+
+BGP and "routing policy" related augmented and not-supported fields are available in openconfig-bgp-ext.yang and openconfig-routing-policy-ext.yang files respectively.
+Supported YANG containers:
+```
+module: openconfig-network-instance
+    +--rw network-instances
+       +--rw network-instance* [name]
+          +--rw table-connections
+          |    ...
+          +--rw protocols
+             +--rw protocol* [identifier name]     
+                +--rw bgp
+                   +--rw global
+                   |  |   ...
+                   |  +--rw afi-safis
+                   |     +--rw afi-safi* [afi-safi-name]
+                   |        ...
+                   +--rw neighbors
+                   |  +--rw neighbor* [neighbor-address]
+                   |     |   ...
+                   |     +--rw afi-safis
+                   |        +--rw afi-safi* [afi-safi-name]
+                   |             ...   
+                   +--rw peer-groups
+                   |  +---rw peer-group* [peer-group-name]
+                   |      |  ...
+                   |      +--rw afi-safis
+                   |         +--rw afi-safi* [afi-safi-name]
+                   |                ...            
+                   +--ro rib
+                      +--ro afi-safis
+                             ...
+
+module: openconfig-routing-policy
+    +--rw routing-policy
+       +--rw defined-sets
+       |     ...
+       +--rw policy-definitions
+          +--rw policy-definition* [name]
+                ...                                             
+
+```
+### 3.6.2 CLI
+  1. For all configuration commands, the CLI request is converted to a corresponding REST client SDK request based on the Open Config data model that was generated by the Swagger generator, and is given to the REST server.
+
+  2. From there on it will follow the same path as a REST config request for create, update and delete operations.
+
+  3. The Swagger generated REST server handles all the REST requests from the client SDK and invokes a common handler for all the create, update, replace, delete and get operations along with path and payload. This common handler converts all the requests into Transformer arguments and invokes the corresponding Transformer APIs.
+
+  4. For show commands, the CLI request is converted to a corresponding REST client SDK get request based on Open Config data model's config or state object on a case by case basis.
+
+  5. For show commands that requires retrieval of the data that doesn't contain any state information (information only based on the configuration), the backend callback will fetch the data from CONFIG_DB.
```

Comment:
what if configuration was changed by frr or vtysh?

Reply:
The recommendation is to use SONIC NBIs (CLI/REST client) for BGP configs, if user does some BGP configs in VTYSH, we may expect to see additional information in the show command.

Reply:
But we should be ready for this. The user 100% will change the configuration of the frr. We must be ready for that and have strategy how to work with it.

Reply:
We're aware of this, we recommend the customers not to use VTYSH for BGP configs, if customers use the both VTYSH and SONiC CLI, it's upto customer to resolve the config. conflicts if any.